### ISSUE TITLE
no-useless-escape should check template literals

### ISSUE THREAD

**Issue by kaicataldo** [eslint/eslint#7331]
`no-useless-escape` does not currently check template literals.

**What version are you using?**
v3.7.1

**What did you do?**
```json
{
  "parserOptions": {
    "ecmaVersion": 6
  },
  "rules": {
    "no-useless-escape": "error"
  }
}
```
```js
var test = `\"${a}\"`;
```
**What happened?**
No errors reported.

**What did you expect to happen?**
```
  1:13  error  Unnecessary escape character: \"  no-useless-escape
  1:19  error  Unnecessary escape character: \"  no-useless-escape

✖ 2 problems (2 errors, 0 warnings)
```

I can put together a PR to fix this.
